app/jobs/simulate_job.rb

```diff
@@ -3,7 +3,7 @@
 class SimulateJob < ApplicationJob
   queue_as :default
 
-  def perform(sim)
-    sim.run
+  def perform(id)
+    Simulation.find(id).run
```

Comment:
Note that this will hard error if `find` fails. In sidekiq, I believes that mean it will retry with exponential backoff.

If the record is truly gone, then sidekiq will waste a time retrying.

If this were a mature production app, I'd say there are two options here:

1. Override the retry behavior to not retry on a missing error, and maybe log the error somewhere
2. Use `Simulation.find_by(id: id)&.run` which will only run this if the record exists, and silently do nothing if the record is gone.

Reply:
I think I can get used to this way of addressing null references. :smile:

Reply:
Wow... when I added your suggestion the linter reduced it to: `Simulation.find_by(id:)&.run` 

It is scary the level of inference that raises the need of Ruby expertise needed to read code.